File: RailgunVB/Commands/Fun/Roll.vb
Imports Discord
Imports Discord.Commands
Imports RailgunVB.Core.Utilities

Namespace Commands.Fun
    
    <Group("roll")>
    Public Class Roll
        Inherits ModuleBase
        
        Private ReadOnly _commandUtils As CommandUtils

        Public Sub New(commandUtils As CommandUtils)
            _commandUtils = commandUtils
        End Sub
        
        <Command>
        Public Async Function RollAsync(Optional num1 As Integer = 0, Optional num2 As Integer = 100) As Task
            Dim rand As New Random
            Dim rng = 0
            
            If num1 > num2
                rng = rand.Next(num2, num1)
            Else 
                rng = rand.Next(num1, num2)
            End If
            
            Dim name As String = await _commandUtils.GetUsernameOrMentionAsync(Context.User)
            await ReplyAsync($"{Format.Bold(name)} has rolled {Format.Bold(rng.ToString())}.")
        End Function
    End Class
    
End NameSpace

File: Railgun/Commands/Fun/Roll.cs
using System;
using System.Threading.Tasks;
using Discord;
using Finite.Commands;
using Railgun.Core.Commands;
using Railgun.Core.Utilities;

namespace Railgun.Commands.Fun
{
    [Alias("roll")]
    public class Roll : SystemBase
    {
        private readonly CommandUtils _commandUtils;

        public Roll(CommandUtils commandUtils) => _commandUtils = commandUtils;
        
        [Command]
        public async Task RollAsync(int num1 = 0, int num2 = 100) {
            var name = await _commandUtils.GetUsernameOrMentionAsync((IGuildUser)Context.Author);
            var rand = new Random();
            var rng = num1 > num2 ? rand.Next(num2, num1) : rand.Next(num1, num2);
            
            await ReplyAsync($"{Format.Bold(name)} has rolled {Format.Bold(rng.ToString())}.");
        }
    }
}